server/filestore_test.go

```diff
@@ -6512,6 +6512,41 @@ func TestFileStoreLargeFullStateMetaCleanup(t *testing.T) {
 
 }
 
+func TestFileStoreIndexDBExistsAfterShutdown(t *testing.T) {
+	sd := t.TempDir()
+	fs, err := newFileStore(
+		FileStoreConfig{StoreDir: sd},
+		StreamConfig{Name: "zzz", Subjects: []string{">"}, Storage: FileStorage})
+	require_NoError(t, err)
+
+	subj, _ := "foo.bar.baz", bytes.Repeat([]byte("ABC"), 33) // ~100bytes
+	for i := 0; i < 1000; i++ {
+		fs.StoreMsg(subj, nil, nil)
+	}
+
```

Comment:
Maybe when you set dirty to true, also remove the existing one. Then you would just need to check for existence.